cpp/src/sort/sort_column.cu

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2020-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sort/sort_impl.cuh>
+
+namespace cudf {
+namespace detail {
+namespace {
+
+/**
+ * @brief Type-dispatched functor for sorting a single column.
+ */
+struct column_sorted_order_fn {
+  /**
+   * @brief Sorts fixed-width columns using faster thrust sort.
+   *
+   * @param input Column to sort
+   * @param indices Output sorted indices
+   * @param ascending True if sort order is ascending
+   * @param stream CUDA stream used for device memory operations and kernel launches
+   */
+  template <typename T, typename std::enable_if_t<cudf::is_fixed_width<T>()>* = nullptr>
+  void faster_sort(column_view const& input,
+                   mutable_column_view& indices,
+                   bool ascending,
+                   rmm::cuda_stream_view stream)
+  {
+    // A non-stable sort on a fixed-width column with no nulls will use a radix sort
+    // if using only the thrust::less or thrust::greater comparators but also
+    // requires making a copy of the input data.
+    auto temp_col = column(input, stream);
+    auto d_col    = temp_col.mutable_view();
+    using DeviceT = device_storage_type_t<T>;
+    if (ascending) {
+      thrust::sort_by_key(rmm::exec_policy(stream),
+                          d_col.begin<DeviceT>(),
+                          d_col.end<DeviceT>(),
+                          indices.begin<size_type>(),
+                          thrust::less<DeviceT>());
```

Comment:
I would think that providing any sort of comparator, even `thrust::less` would result in using the merge sort path. Have you verified it will still take the radix sort path for appropriate types?

Reply:
Yes, `thrust::sort` has specialization that allows for `thrust::less` and `thrust::greater` comparators only.
It will use the `radix_sort` path if `enable_if_primitive_sort` is true which is determined by
https://github.com/NVIDIA/thrust/blob/4fd1b54cece96c56e49d6a3fc8df6c4ab1c9499c/thrust/system/cuda/detail/sort.h#L1510-L1522

Also, the benchmark numbers were still about the same.

Reply:
huh, neat. Thanks for confirming.